core/interop/comwrappers/IDispatch/ComWrappersImpl.cs

```diff
@@ -13,31 +13,24 @@ internal unsafe class ComWrappersImpl : ComWrappers
     private static readonly ComInterfaceEntry* wrapperEntry;
 
     // This class only exposes IDispatch and the vtable is always the same.
-    // The below isn't the most efficient but it is reasonable for prototyping.
     // If additional interfaces want to be exposed, add them here.
     static ComWrappersImpl()
     {
-        GetIUnknownImpl(out IntPtr fpQueryInteface, out IntPtr fpAddRef, out IntPtr fpRelease);
-
-        var vtbl = new IDispatchVtbl()
-        {
-            IUnknownImpl = new IUnknownVtbl()
-            {
-                QueryInterface = fpQueryInteface,
-                AddRef = fpAddRef,
-                Release = fpRelease
-            },
-            GetTypeInfoCount = Marshal.GetFunctionPointerForDelegate(IDispatchVtbl.pGetTypeInfoCount),
-            GetTypeInfo = Marshal.GetFunctionPointerForDelegate(IDispatchVtbl.pGetTypeInfo),
-            GetIDsOfNames = Marshal.GetFunctionPointerForDelegate(IDispatchVtbl.pGetIDsOfNames),
-            Invoke = Marshal.GetFunctionPointerForDelegate(IDispatchVtbl.pInvoke)
-        };
-        var vtblRaw = RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IDispatchVtbl), sizeof(IDispatchVtbl));
-        Marshal.StructureToPtr(vtbl, vtblRaw, false);
-
-        wrapperEntry = (ComInterfaceEntry*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IDispatchVtbl), sizeof(ComInterfaceEntry));
-        wrapperEntry->IID = typeof(IDispatch).GUID;
-        wrapperEntry->Vtable = vtblRaw;
+        GetIUnknownImpl(out IntPtr fpQueryInterface, out IntPtr fpAddRef, out IntPtr fpRelease);
```

Comment:
I was suggested to use `GetIUnknownImpl(out vtblRaw[0], out vtblRaw[1], out vtblRaw[2]);` to reduce boilerplate previously in related discussions. Is this advice changed or here is inappropriate to show that to others?

Reply:
I can change it to that.

Reply:
Actually after looking at that I think I am going to keep it as is. The main reason is if someone adds another interface to support calling `GetIUnknownImpl` isn't needed and so will need to be teased out to reuse the pointers.